dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilderUtility.h

```diff
@@ -96,26 +78,41 @@ inline PCWSTR GetWinSoundEventString(AppNotificationBuilder::AppNotificationSoun
     }
 }
 
-inline std::wstring EncodeArgument(std::wstring const& value)
+inline std::wstring GetArgumentEncoding(wchar_t const& ch)
 {
-    std::wstring encodedValue{};
+    static std::unordered_map<wchar_t, std::wstring> encodings = { { L'&', L"&amp;"}, { L'\"', L"&quot;"}, {L'<', L"&lt;"},
+                                                                        {L'>', L"&gt;"}, {L'\'', L"&apos;"}, { L'%', L"%25"},
+                                                                            {L';', L"%3B"}, {L'=', L"%3D"} };
+    if (encodings.find(ch) != encodings.end())
+    {
+        return encodings[ch];
+    }
+    else
+    {
+        return std::wstring(1, ch);
+    }
+}
 
-    auto percentEncodings{ GetPercentEncodings() };
-    auto xmlEncodings{ GetXmlEscapeEncodings() };
+inline std::wstring GetXmlEncoding(wchar_t const& ch)
+{
+    static std::unordered_map<wchar_t, std::wstring> encodings = { { L'&', L"&amp;"}, { L'\"', L"&quot;"}, {L'<', L"&lt;"},
+                                                                        {L'>', L"&gt;"}, {L'\'', L"&apos;"} };
+    if (encodings.find(ch) != encodings.end())
+    {
+        return encodings[ch];
+    }
+    else
+    {
+        return std::wstring(1, ch);
+    }
+}
+
+inline std::wstring EncodeArgument(winrt::hstring const& value)
+{
+    std::wstring encodedValue{};
```

Comment:
Should we assign a capacity here for the string so that it doesn''t have to guess? What is the max expected length here?

Reply:
I'm not sure what the max is here, the only max we are enforcing the general AppNotification size.